### PR TITLE
Adding Create Dryrun sdk doc to debugging

### PR DESCRIPTION
**Pull request by barnjamin**

### CODE REVIEW

File: docs/get-details/dapps/smart-contracts/debugging.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+# Using the TEAL debugger
+Algorand provides the `tealdbg` command-line tool to launch an interactive session to debug smart contracts. This tool is explained in the project’s [README](https://github.com/algorand/go-algorand/blob/master/cmd/tealdbg/README.md).
+
+This debugger can debug local smart contracts or connect remotely to an on-chain smart contract. The examples below illustrate using the debugger locally, which will be the predominant use case for developers when they are developing smart contracts. For more information on the `tealdbg` utility and how you can use it remotely, see the project’s [README](https://github.com/algorand/go-algorand/blob/master/cmd/tealdbg/README.md) which covers in detail the different debugging options.
+
+The debugger process supports both Chrome Developer Tools and a simple Web Frontend.
```
Comment (ryanRfox) [inline docs/get-details/dapps/smart-contracts/debugging.md:10]: Add "(CDT)" after Chrome Developer Tools.

File: docs/get-details/dapps/smart-contracts/debugging.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    FileOutputStream outfile = new FileOutputStream(fname);
+    outfile.write(Encoder.encodeToMsgPack(drr));
+    outfile.close();
+
+    ```
+
+=== "goal"
+    ```sh
+
+    $ goal app call --app-id {appid} --from {ACCOUNT} --out=dryrun.msgp --dryrun-dump
```
Comment (ryanRfox) [inline docs/get-details/dapps/smart-contracts/debugging.md:131]: I thought default was json format. If true, change file extension to .json

File: docs/get-details/dapps/smart-contracts/debugging.md [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+To learn more about setting individual context items, see the `tealdbg` [documentation](https://github.com/algorand/go-algorand/blob/master/cmd/tealdbg/README.md).
+
+The context can also be read directly from an instance of the Indexer. See the `tealdbg` [documentation](https://github.com/algorand/go-algorand/blob/master/cmd/tealdbg/README.md) for more information.
+
+
+## Calling the debugger with context
+
+```
+$ tealdbg debug -d dryrun.msgp
```
Comment (ryanRfox) [inline docs/get-details/dapps/smart-contracts/debugging.md:158]: match file extension here if changed above.

**Review by ryanRfox — CHANGES_REQUESTED** [review]
A few items for consideration

File: docs/get-details/dapps/smart-contracts/debugging.md [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+$ tealdbg debug program.teal
+.
+.
+2020/08/25 14:05:38 CDT debugger listening on: ws://localhost:9392/091d04a69152223ae84c8b40271c3d62f8490ea9b795aae95868932163f89735
+2020/08/25 14:05:38 Or open in Chrome:
+2020/08/25 14:05:38 devtools://devtools/bundled/js_app.html?experiments=true&v8only=false&ws=localhost:9392/091d04a69152223ae84c8b40271c3d62f8490ea9b795aae95868932163f89735
+```
+
+This will launch the debugger process and return an endpoint that is listening for connections. This process can be connected to directly with the Chrome Developer Tools. The simplest way to do this is to enter `chrome://inspect/` into the address bar of the browser, click “Configure” to add “localhost:9392”, and select the Algorand TEAL Debugger in the Remote Target Section (click on the inspect link).
+
```
[the remark this answers is not in the document]
Comment (barnjamin) [reply docs/get-details/dapps/smart-contracts/debugging.md:24]: this has tripped me up a few times, I cant recall exactly why but i found it unreliable (port and ip maybe?). the chrome://inspect should always show the correct url though

**Pull request merged**